solutions/USACO open20-haircut.mdx

```diff
@@ -0,0 +1,167 @@
+---
+id: usaco-1041
+source: USACO Gold 2020 US Open
+title: Haircut
+author: Jeffrey Meng
+---
+
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_haircut_gold_open20.html)
+
+First, we need to count the number of inversions. (Fun fact: the number of inversions is also a measure of how sorted an array is. If an array has 0 inversions, it's perfectly sorted in ascending order.) 
+
+One way to count inversions is to use a Binary Indexed Tree. Think of an inversion as a pair $i, j$, with $i < j$ & $A[i] > A[j]$. Then, we can easily compute, for each element in the array, the number of inversions it is $j$ for.
```

Comment:
Why is "Binary Indexed Tree" capitalized? Also, an inversion **is** a pair $i, j$ such that ...